EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -162,30 +153,31 @@ void setContactorsTask(void const* arg)
 
                 auxStatus.contactorError = contactorError;
 
-                if (hasDischargeBeenSet && hasChargeBeenSet)
+                if (dischargeSense && chargeSense)
                 {
                     auxStatus.startUpSequenceDone = 1;
-                    state = DONE;
                 }
 
                 osMutexRelease(auxStatus.auxStatusMutex);
 
-                if (!hasDischargeBeenSet)
+                if (dischargeSense && chargeSense)
                 {
-                    HAL_GPIO_WritePin(DISCHARGE_ENABLE_GPIO_Port, DISCHARGE_ENABLE_Pin, GPIO_PIN_SET);
-                    prevState = state;
-                    state = DISCHARGE_ENABLE_CHECK;
+                    state = DONE;
                 }
-                else if (!hasChargeBeenSet)
+                else // Alternate between CHARGE_ENABLE_CHECK & DISCHARGE_ENABLE_CHECK
                 {
                     prevState = state;
                     state = DISCHARGE_ENABLE_CHECK;
+
+                    if (!dischargeSense)
+                    {
+                        HAL_GPIO_WritePin(DISCHARGE_ENABLE_GPIO_Port, DISCHARGE_ENABLE_Pin, GPIO_PIN_SET);
+                    }
                 }
 
                 uint8_t commonSense = !HAL_GPIO_ReadPin(COMMON_SENSE_GPIO_Port, COMMON_SENSE_Pin);
-                uint8_t dischargeSense = !HAL_GPIO_ReadPin(DISCHARGE_SENSE_GPIO_Port, DISCHARGE_SENSE_Pin);
 
-                if (!commonSense || (hasDischargeBeenSet && !dischargeSense && !orionStatus.dischargeContactorOverriden))
+                if (!commonSense || (!dischargeSense && !orionStatus.dischargeContactorOverriden))
```

Comment:
This may not work under normal operation. The sequence is Common -> Charge, so at this point, discharge wouldn't be set, so I think the contactors would get disconnected every time. If think we would have to check that the previous state was `DISCHARGE_ENABLE_CHECK` or `DONE`?

Reply:
That would work for DONE, but not if the discharge check finished with the discharge contactor not closing.... maybe we do need to keep track of that?

Reply:
Hmmm. The other question is do we want to disconnect the contactors if charge or discharge is open but should be closed?

Reply:
Don't we do that in done?

Reply:
Yeah,  maybe we only do it in done?